Datastore/src/DatastoreClient.php

```diff
@@ -530,12 +543,14 @@ public function allocateIds(array $keys, array $options = [])
      *
      *     @type array $transactionOptions Transaction configuration. See
      *           [ReadWrite](https://cloud.google.com/datastore/docs/reference/rest/v1/projects/beginTransaction#ReadWrite).
+     *     @type string $databaseId The value of databaseId to be used for queries.
      * }
      * @return Transaction
      * @codingStandardsIgnoreEnd
      */
     public function transaction(array $options = [])
     {
+        $options += ['databaseId' => ''];
         $transaction = $this->operation->beginTransaction([
```

Comment:
In order to make sure the databaseId set here takes precedent over one that may be set at the constructor level of this class we'll need to update how `Operation::beginTransaction` / `Operation::allocateIds` / `Operation::lookup` merges the options array together.

e.g.,

```diff
- $res = $this->connection->beginTransaction([
+ $res = $this->connection->beginTransaction($options + [
    'projectId' => $this->projectId,
    'databaseId' => $this->databaseId,
    'transactionOptions' => $transactionOptions,
- ] + $options);
+ ]);
```

Reply:
Would you be open to adding some unit tests to help cover these cases, too? To make sure the ID set here takes precedent over any existing one in the constructor?

Reply:
I added some tests in `OperationsTest.php` by the name `*WithDatabaseIdOverride`.